modules/bibcheck/lib/plugins/remove_subfield_matching.py

```diff
@@ -0,0 +1,49 @@
+# -*- coding: utf-8 -*-
+##
+## This file is part of Invenio.
+## Copyright (C) 2015 CERN.
+##
+## Invenio is free software; you can redistribute it and/or
+## modify it under the terms of the GNU General Public License as
+## published by the Free Software Foundation; either version 2 of the
+## License, or (at your option) any later version.
+##
+## Invenio is distributed in the hope that it will be useful, but
+## WITHOUT ANY WARRANTY; without even the implied warranty of
+## MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+## General Public License for more details.
+##
+## You should have received a copy of the GNU General Public License
+## along with Invenio; if not, write to the Free Software Foundation, Inc.,
+## 59 Temple Place, Suite 330, Boston, MA 02111-1307, USA.
+
+""" Bibcheck plugin to remove a subfield matching regexp """
+
+import re
+
+def check_record(record, subfield=None, regexp=None):
+    """
+    remove specific subfield matching anchored regexp
+    if the matching subfield is the only one in the MARC tag,
+    remove the entire MARC tag
+    """
+    if subfield is None or regexp is None:
+        return
+    try:
+        subfieldre = re.compile(regexp)
+    except re.error:
+        return
+    for pos, val in record.iterfield(subfield):
+        if subfieldre.match(val):
+            count = 0
+            for spos, _ in record.iterfield(subfield[0:5] + '%'):
```

Comment:
in the end:

    count  = len(record[pos[0][0:3]][pos[1]][0])

and wouldn't it be better if bibcheck leans to delete empty subfields?